README.md

```diff
@@ -51,85 +51,125 @@ some kind of discriminant:
 
 # Syntax
 
+## Enum Declarations
+
 ```ts
-// enum declarations
+// Enum Declarations
 
-// Each auto-initialized member value is a `Number`, auto-increments values by 1 starting at 0
-enum Numbers {
-  zero,
-  one,
-  two,
-  three,
-  alsoThree = three
+// Enum without `of` clause requires explicit initializers.
+enum Explicit {
+  zero = 0,
+  abc = "ABC",
+  sym = Symbol(),
 }
 
-// Each auto-initialized member value is a `Number`, auto-increments values by 1 starting at 0
+// Enum member names can be valid property names:
+enum Named {
+  identifierName = 0,
+  "string name" = 1,
+  [expr] = 2,
+}
+
+// `of` clause:
+
+// Each auto-initialized member value is a `Number`, auto-increments values by 1 starting at 0.
 enum Colors of Number {
   red,
   green,
-  blue
+  blue,
 }
 
 // Each auto-initialized member value is a `String` whose value is the SV of its member name.
 enum PlayState of String {
   idle,
   running,
-  paused
+  paused,
 }
 
 // Each auto-initialized member value is a `Symbol` whose description is the SV of its member name.
 enum Symbols of Symbol {
   alpha,
-  beta
+  beta,
 }
 
-enum Named {
-  identifierName,
-  "string name",
-  [expr]
+// A special `Enum.ADT` mapper exists that can be used to define algebraic data types:
+enum Message of Enum.ADT {
```

Comment:
I wonder how this dynamic dispatch approach (delegate to Enum.ADT[@@toEnum]) works with TypeScript.

Reply:
It may not matter too much for native `enum` due to differences in how we handle `enum` vs. `const enum`. For something like `enum Color of Number { red, green, blue }`, default members would be `TypeFlags.EnumLiteral | TypeFlags.Number`.

I'm currently iterating on the ADT enum approach. I'm considering the possibility of having the `of` mapper only apply to _Scalar_ members that don't use ADT enum syntax, and having ADT enum members (that use `{}` or `()`) always produce an ADT enum member factory. Essentially, that would mean that `of` would only apply to uninitialized scalar values.